openshift-knative-operator/cmd/operator/kodata/knative-serving/0.19.0/7-rbac-proxy.yaml

```diff
@@ -0,0 +1,293 @@
+  namespaceSelector:
+    matchNames:
+    - knative-serving
+  selector:
+    matchLabels:
+      name: activator-sm-service
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: autoscaler-sm
+  namespace: knative-serving
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      bearerTokenSecret:
+        key: ""
+      port: https
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+      - knative-serving
+  selector:
+    matchLabels:
+      name: autoscaler-sm-service
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: autoscaler-hpa-sm
+  namespace: knative-serving
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      bearerTokenSecret:
+        key: ""
+      port: https
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+      - knative-serving
+  selector:
+    matchLabels:
+      name: autoscaler-hpa-sm-service
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: controller-sm
+  namespace: knative-serving
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      bearerTokenSecret:
+        key: ""
+      port: https
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+      - knative-serving
+  selector:
+    matchLabels:
+      name: controller-sm-service
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: domain-mapping-sm
+  namespace: knative-serving
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      bearerTokenSecret:
+        key: ""
+      port: https
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+      - knative-serving
+  selector:
+    matchLabels:
+      name: domain-mapping-sm-service
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: domainmapping-webhook-sm
+  namespace: knative-serving
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      bearerTokenSecret:
+        key: ""
+      port: https
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+      - knative-serving
+  selector:
+    matchLabels:
+      name: domainmapping-webhook-sm-service
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: webhook-sm
+  namespace: knative-serving
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      bearerTokenSecret:
+        key: ""
+      port: https
+      scheme: https
+      tlsConfig:
+        insecureSkipVerify: true
```

Comment:
If we're skipping the verification everywhere, should we just serve HTTP from the RBAC proxy?

Reply:
This is for the ca verification, we want https to encrypt stuff. I will check if I can remove that.

Reply:
well, skipping CA verification opens the door for MITM attacks, which generally renders the encryption useless.

Reply:
@markusthoemmes how do we generate certificates for the rest of our stuff maybe we can re-use that.

Reply:
Btw this is not a blocker unless we dont trust the rbac-proxy which is too much I think.

Reply:
Sure it's not a blocker, I was just wondering why we don't just do plain HTTP and spare us the extra config

Reply:
If your certs are generated by the internal service CA operator (generated via annotations) then we have the CA available in the client (prometheus). See https://github.com/openshift/cluster-monitoring-operator/blob/6ba1db4ca73eca18c28acef3b865a892fa26e719/assets/alertmanager/service-monitor.yaml#L14-L16 as an example (just random).

You just have to set the serverName of your TLS kube-rbacy-proxy endpoint.

Putting my SRE hat on: For me (personally) this should be a blocker if it is fixable in here.

Reply:
And this is the annotation I talked about above: https://github.com/openshift/cluster-monitoring-operator/blob/6ba1db4ca73eca18c28acef3b865a892fa26e719/assets/alertmanager/service.yaml#L4-L5

It will create server public/private key in a secret called `alertmanager-main-tls`.

Reply:
For the purpose of this PR, it is fine with me to delay the HTTPS stuff to an immediate followup. We should at least record it in JIRA and make it a blocker for 1.13 though.

Reply:
@markusthoemmes plain http does not provide encryption. Better secure it end-to-end imho.

Reply:
With the service CA operator mentioned by @s-urbaniak

Reply:
Ok let me try and see if I can make it work and update the PR.

Reply:
@s-urbaniak I dont think the secret is created when I needed do I have to wait for it? Right now we have some yaml files that we apply in order to create the service monitors, services and the deployments.
Also do I need to set these args https://github.com/openshift/cluster-monitoring-operator/blob/6ba1db4ca73eca18c28acef3b865a892fa26e719/assets/alertmanager/alertmanager.yaml#L66-L67 when using the kube-rbac-proxy so that it uses the certificate that is mounted?

Reply:
@s-urbaniak is it ok not to use a serverName in the service monitor tlsconfig, I am getting 
```
Get "https://10.116.0.94:8444/metrics": x509: certificate is valid for autoscaler-hpa-644c88686d-cm8zn, not autoscaler-hpa-sm-service.knative-serving.svc
```
Reason is I dont know the name of the pods behind a deployment and they are not static.

Reply:
Figured it out will update the PR shortly, I was getting the default certs instead of the service ones.